syntax = "proto3";

package suggestions;

service SuggestionsService {
    rpc GetSuggestions(SuggestionRequest) returns (SuggestionResponse);
}

message SuggestionRequest {
    repeated string purchased_books = 1;
}

message SuggestionResponse {
    repeated string suggested_books = 1;
    map<string, int32> vector_clock = 2; // ðŸ†• Add this to support logging and concurrency tracking
}
